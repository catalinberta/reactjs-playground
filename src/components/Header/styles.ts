import styled from 'styled-components';
import LogoWhiteImage from '@src/static/assets/images/logo-white.png';

export const Wrapper = styled.div`
	width: 100%;
	height: 100%;
	display: flex;
	justify-content: center;
	background-color: ${(props) => props.theme.colors.primary.main};
	color: #fff;
`;

export const Container = styled.div(
	({ theme }) => `
  	width: 100%;
	max-width: 1200px;
	display: flex;
	justify-content: space-between;
	align-items: center;
	${theme.responsive(theme.breakpoints.uhd)} {
		max-width: 2000px;
	}
`
);

export const Logo = styled.div`
	width: 50px;
	height: 50px;
	background-size: contain;
	background-image: url(${LogoWhiteImage});
	background-repeat: no-repeat;
`;
