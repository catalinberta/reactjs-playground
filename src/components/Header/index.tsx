import modules from '@src/constants/modules';
import React from 'react';
import { useHistory } from 'react-router-dom';
import DropdownMenu from '../DropdownMenu';
import { Container, Logo, Wrapper } from './styles';

const Header = (): React.ReactElement => {
	const history = useHistory();

	const onModuleChange = (value: any) => {
		history.push(modules[value].routePath);
	};

	return (
		<Wrapper>
			<Container>
				<Logo data-testid="logo" />
				<DropdownMenu onChange={onModuleChange} />
			</Container>
		</Wrapper>
	);
};

export default Header;
