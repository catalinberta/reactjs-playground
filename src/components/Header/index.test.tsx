import React from 'react';
import { render } from '@testing-library/react';
import Header from './';
import renderComponent from '../../tests/ComponentRenderer';

describe('Header Component', () => {
	test('Snapshot', () => {
		const { asFragment } = render(renderComponent(<Header />));

		expect(asFragment()).toMatchSnapshot();
	});

	test('Renders Logo', () => {
		const { getByTestId } = render(renderComponent(<Header />));
		const headerLogo = getByTestId('logo');

		expect(headerLogo).toBeDefined();
	});
});
