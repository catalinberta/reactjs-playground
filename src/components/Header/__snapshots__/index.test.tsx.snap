// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz hWKGQE"
  >
    <div
      class="sc-dlfnbm imUggJ"
    >
      <div
        class="sc-hKgILt fwgyoG"
        data-testid="logo"
      />
      <div
        class="MuiFormControl-root-4 sc-bdfBwQ huyxwo makeStyles-formControl-1"
      >
        <label
          class="MuiFormLabel-root-16 MuiInputLabel-root-8 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiFormLabel-filled-18"
          data-shrink="true"
          id="menuItemLabel"
        >
          Choose playground
        </label>
        <div
          class="MuiInputBase-root-41 MuiInput-root-30 MuiInput-underline-33 makeStyles-selectedEmpty-2 MuiInputBase-formControl-42 MuiInput-formControl-31"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root-20 MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-49 MuiInput-input-37"
            role="button"
            tabindex="0"
          >
            Redux Classic
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput-29"
            tabindex="-1"
            value="reduxClassic"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-56 MuiSelect-icon-25"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
